src/sql/workbench/contrib/query/browser/queryInputFactory.ts

```diff
@@ -83,7 +83,9 @@ export class QueryEditorLanguageAssociation implements ILanguageAssociation {
 		}
 
 		const profile = getCurrentGlobalConnection(this.objectExplorerService, this.connectionManagementService, this.editorService);
```

Comment:
Make the existing profile check into an outer if and then only get the global connection if we don't have an existing one currently.

Make sure you thoroughly test opening queries as well - especially when untitled queries are re-used (new untitled query -> change connection to a different db -> close query window -> new untitled query again which should re-use the same name)

Reply:
It doesn't seem to call the syncConvertInput neither is the convertInput whatsoever, so this does not affect untitledQueries it seems

Reply:
also convertInput is never called, so I'm not sure if I should add the existing check for it as well, wouldn't hurt though.